.evergreen/aws_lambda/run-deployed-lambda-aws-tests.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+set -o errexit  # Exit the script with error if any of the commands fail
+
+# Explanation of required environment variables:
+#
+# TEST_LAMBDA_DIRECTORY: The root of the project's Lambda sam project.
+# DRIVERS_ATLAS_PUBLIC_API_KEY: The public Atlas key for the drivers org.
+# DRIVERS_ATLAS_PRIVATE_API_KEY: The private Atlas key for the drivers org.
+# DRIVERS_ATLAS_LAMBDA_USER: The user for the lambda cluster.
+# DRIVERS_ATLAS_LAMBDA_PASSWORD: The password for the user.
+# DRIVERS_ATLAS_GROUP_ID: The id of the individual projects under the drivers org, per language.
+# LAMBDA_STACK_NAME: The name of the stack on lambda "dbx-<language>-lambda"
+# AWS_REGION: The region for the function - generally us-east-1
+
+# Explanation of generated variables:
+#
+# MONGODB_URI: The URI for the created Atlas cluster during this script.
+# FUNCTION_NAME: Uses the stack name plus the current commit sha to create a unique cluster and function.
+# CREATE_CLUSTER_JSON: The JSON used to create a cluster via the Atlas API.
+# ATLAS_BASE_URL: Where the Atlas API root resides.
+
+# The base Atlas API url. We use the API directly as the CLI does not yet
+# support testing cluster outages.
+ATLAS_BASE_URL="https://cloud.mongodb.com/api/atlas/v1.0"
+
+# Add git commit to name of function and cluster.
+FUNCTION_NAME="${LAMBDA_STACK_NAME}-$(git rev-parse --short HEAD)"
+
+# Check is cluster has a srv address, and assume once it does, it can be used.
+check_cluster ()
+{
+  count=0
+  SRV_ADDRESS="null"
+  # Don't try longer than 15 minutes.
+  while [ $SRV_ADDRESS = "null" ] && [ $count -le 30 ]; do
+    echo "Checking every 30 seconds for cluster to be created..."
+    # Poll every 30 seconds to check the cluster creation.
+    sleep 30
+    SRV_ADDRESS=$(curl \
+      --digest -u "${DRIVERS_ATLAS_PUBLIC_API_KEY}:${DRIVERS_ATLAS_PRIVATE_API_KEY}" \
+      -X GET \
+      "${ATLAS_BASE_URL}/groups/${DRIVERS_ATLAS_GROUP_ID}/clusters/${FUNCTION_NAME}" \
+      | jq -r '.srvAddress'
+    );
+    count=$(( $count + 1 ))
+    echo $SRV_ADDRESS
+  done
+
+  if [ $SRV_ADDRESS = "null" ]; then
+    echo "No cluster could be created in the 15 minute timeframe or error occured."
+    exit 1
+  else
+    echo "Setting MONGODB_URI in the environment to the new cluster."
+    # else set the mongodb uri
+    URI=$(echo $SRV_ADDRESS | grep -Eo "[^(\/\/)]*$" | cat)
+    MONGODB_URI="mongodb+srv://${DRIVERS_ATLAS_LAMBDA_USER}:${DRIVERS_ATLAS_LAMBDA_PASSWORD}@${URI}"
+    export MONGODB_URI=$MONGODB_URI
+  fi
+}
```

Comment:
Does this logic belong in the setup script instead?  I'd expect the setup script to fail if its unable to provision an atlas cluster in the time limit (15 minutes).  

I'm thinking of reusing the setup/teatdown (with potentially some slight modifications) for the search indexes testing, but having the `check_cluster` logic inside the script to run the lambda tests makes it hard to share the logic.

Reply:
Oh I'm still in draft here. :) I was planning on moving it to the setup script - just needed to get it working then figure out the best way to get the MONGODB_URI passed through.

Reply:
Sorry to step on your toes during draft phase, should have looked more closely at the PR before commenting 😬

Reply:
Updated to move the check into the setup and write the URI to a YML file for expansion usage in evg.